WordPress/Classes/Services/ReaderPostService.m

```diff
@@ -435,6 +437,63 @@ - (void)toggleSavedForLaterForPost:(ReaderPost *)post
     }];
 }
 
+- (void)toggleSeenForPost:(ReaderPost *)post
+                  success:(void (^)(void))success
+                  failure:(void (^)(NSError *error))failure;
+{
+    [self.managedObjectContext performBlock:^{
+        
+        // Get a the post in our own context
+        NSError *error;
+        ReaderPost *readerPost = (ReaderPost *)[self.managedObjectContext existingObjectWithID:post.objectID error:&error];
+        if (error) {
+            dispatch_async(dispatch_get_main_queue(), ^{
+                if (failure) {
+                    failure(error);
+                }
+            });
+            return;
+        }
+        
+        // Keep previous values in case of failure
+        BOOL oldValue = readerPost.isSeen;
+        BOOL seen = !oldValue;
+        
+        // Optimistically update
+        readerPost.isSeen = seen;
+        [[ContextManager sharedInstance] saveContext:self.managedObjectContext];
+        
+        // Define success block.
+        void (^successBlock)(void) = ^void() {
+            if (success) {
+                success();
+            }
+        };
+        
+        // Define failure block. Make sure rollback happens in the managedObjectContext's queue.
+        void (^failureBlock)(NSError *error) = ^void(NSError *error) {
+            [self.managedObjectContext performBlockAndWait:^{
+                
+                DDLogError(@"Error while toggling post Seen status: %@", error);
+                readerPost.isSeen = oldValue;
+                
+                [[ContextManager sharedInstance] saveContext:self.managedObjectContext withCompletionBlock:^{
+                    if (failure) {
+                        failure(error);
+                    }
+                }];
+            }];
+        };
+        
+        // Call the remote service.
+        ReaderPostServiceRemote *remoteService = [[ReaderPostServiceRemote alloc] initWithWordPressComRestApi:[self apiForRequest]];
+        [remoteService markPostSeenWithSeen:seen
+                                     feedID:readerPost.feedID
+                                 feedItemID:readerPost.feedItemID
```

Comment:
I was doing some more testing and I'm seeing something new. The value of feedItemID is turning up nil for the sites I've tested.  I haven't dug in to figure out why but it's happened on a private site of mine, an internal p2, and a public blog.   
Just dropping a quick note.  I'll take another look first thing tomorrow.

Reply:
I think what might have happened is I updated my local branch before 73dc7be and 25d4f77 were pushed up.  I've updated and tried a clean build and now things work as expected. :)

Reply:
Sorry! That's what I get for assuming you wouldn't re-review so soon! 😄